contrib/dockerd-rootless.sh

```diff
@@ -139,6 +149,25 @@ if [ -z "$_DOCKERD_ROOTLESS_CHILD" ]; then
 		"$0" "$@"
 else
 	[ "$_DOCKERD_ROOTLESS_CHILD" = 1 ]
+
+	# The Container Device Interface (CDI) specs can be found by default
+	# under {/etc,/var/run}/cdi. More information at:
+	# https://github.com/cncf-tags/container-device-interface
+	#
+	# In order to use the Container Device Interface (CDI) integration,
+	# the CDI paths need to exist before the Docker daemon is started in
+	# order for it to read the CDI specification files. Otherwise, a
+	# Docker daemon restart will be required for the daemon to discover
+	# them.
+	#
+	# If another set of CDI paths (other than the default /etc/cdi and
+	# /var/run/cdi) are configured through the Docker configuration file
+	# (using "cdi-spec-dirs"), they need to be bind mounted in rootless
+	# mode; otherwise the Docker daemon won't have access to the CDI
+	# specification files.
+	[ -d "/etc/cdi" ] && mount_directory /etc/cdi
```

Comment:
Does `-d` really work with `copy-up`?
It is usually a symlink

Reply:
We are not `copy-upping` these directories. They will be read-only by dockerd. Do we still need to do a `copy-up`?

It is working for me as it is, but let me know if I'm missing anything.

Reply:
`/etc` is copied-up so `/etc/cdi` usually becomes a symlink.

I assumed `[ -d ]` could fail for a symlink to a dir, but TIL I was wrong.


> TEST(1)
> [...]
> Except  for -h and -L, all FILE-related tests dereference symbolic links.